### ISSUE TITLE
Game freezes when using any implement with Syn Trac in fieldwork mode

### ISSUE THREAD

**Issue by bovawegav** [Courseplay/courseplay#4490]
I am using the very latest courseplay version and the latest version of Syn Trac. Every implement that has three point connector works fine, but as soon as I attach anything like trailers, loading wagons etc. that only has the hitch and I enter in "fieldwork" mode on courseplay the game freezes. The log gets spammed with this error _"vehicles.lua:897: attempt to index field 'attacherJoint' (a nil value)"_

I didnt test it on the 2x6 wheel version of the syntrac, only the 2x2 with the attacher part on the back. Happens every time. It works when I attach the "attacher" part of the Syn Trac in front, but in that way it is impossible to do any fieldwork.

**Comment by Tensuko** [comment]
I will take a look at it and do some tests to find out whats wrong, but that will take some time.
We already had Issues with the SyncTrac. You would have to not use it for now, sry.

**Comment by bovawegav** [comment]
> I will take a look at it and do some tests to find out whats wrong, but that will take some time.
> We already had Issues with the SyncTrac. You would have to not use it for now, sry.

Ok, no problem. If you do find out what is the problem please post it here, I will be following this thread. But no pressure obviously, since this is a mod-mod conflict this shouldn't be your priority or even something you should try to fix.

**Comment by bovawegav** [comment]
I tested it quite a lot today and I think I might have narrowed down the cause of the problem. I am wiring this down so in case you take a look into this you know exactly where to look. So first of all the only time this occurs is with the 2x2 version of the Syn Trac with the 'standart adapter' attached on the back. The 2x3 version works fine. I assume it is because there is no need for an adapter part for it as the attacher already exists on the base model. 
The problem and game freeze occurs only when any implement is attached on the 2x2 Syn Trac **while** the coourseplay window is open on the 'fieldwork' mode. If the courseplay window is opened on any other tab the game freeze doesn't occur when something is attached to the Syn Trac. Entering the 'fieldwork' mode _after_ attaching implement also doesn't freeze the game and it is possible to generate courses and do actual fieldwork using courseplay. I was also able to recreate this problem on my other laptop which only has the game, syn trac and courseplay loaded.

**Comment by Slebbog** [comment]
Yeah can confirm but I also can't do any fieldwork with the four wheeled Syn Track. I generate a course and click "drive now" but it just stands still and doesn't move and nothing happens. It does hire a worker but the course-play doesn't even understand that I have pressed "drive now", I can simply keep pressing the button. Didn't try the six wheeled version.

**Comment by SteelRodent** [comment]
AFAIK the SynTrac isn't registered as a standard vehicle type. At least the description says it's a new vehicle type. And CP doesn't like non-standard vehicle types.
